core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -67,8 +67,11 @@ private class AsyncEventQueue(
   /** A counter for dropped events. It will be reset every time we log it. */
   private val droppedEventsCounter = new AtomicLong(0L)
 
+  /** A counter to keep number of dropped events last time it was logged */
+  private var lastDroppedEventsCounter: Long = 0L
```

Comment:
should have `@volatile` to safely access this.

Reply:
Good comments, I will update.